flink-filesystems/flink-s3-fs-base/src/test/java/org/apache/flink/fs/s3/common/writer/RecoverableMultiPartUploadImplTest.java

```diff
@@ -123,127 +116,82 @@ public void multiplePartAndObjectUploadsShouldBeReflectedInRecoverable() throws
 
         final S3Recoverable recoverable = uploadObject(thirdIncompletePart);
 
-        assertThat(
-                recoverable, isEqualTo(thirdIncompletePart, firstCompletePart, secondCompletePart));
+        assertThatIsEqualTo(
+                recoverable, thirdIncompletePart, firstCompletePart, secondCompletePart);
     }
 
     @Test
-    public void s3RecoverableReflectsTheLatestPartialObject() throws IOException {
+    void s3RecoverableReflectsTheLatestPartialObject() throws IOException {
         final byte[] incompletePartOne = bytesOf("AB");
         final byte[] incompletePartTwo = bytesOf("ABC");
 
         S3Recoverable recoverableOne = uploadObject(incompletePartOne);
         S3Recoverable recoverableTwo = uploadObject(incompletePartTwo);
 
-        assertThat(
-                recoverableTwo.incompleteObjectName(),
-                not(equalTo(recoverableOne.incompleteObjectName())));
+        assertThat(recoverableTwo.incompleteObjectName())
+                .isNotEqualTo(recoverableOne.incompleteObjectName());
     }
 
-    @Test(expected = IllegalStateException.class)
-    public void uploadingNonClosedFileAsCompleteShouldThroughException() throws IOException {
+    @Test
+    void uploadingNonClosedFileAsCompleteShouldThroughException() throws IOException {
         final byte[] incompletePart = bytesOf("!!!");
 
         final RefCountedBufferingFileStream incompletePartFile = writeContent(incompletePart);
 
-        multiPartUploadUnderTest.uploadPart(incompletePartFile);
+        assertThatThrownBy(() -> multiPartUploadUnderTest.uploadPart(incompletePartFile))
+                .isInstanceOf(IllegalStateException.class);
     }
 
-    // --------------------------------- Matchers ---------------------------------
-
-    private static TypeSafeMatcher<StubMultiPartUploader> hasMultiPartUploadWithPart(
-            final int partNo, final byte[] content) {
-
-        final TestUploadPartResult expectedCompletePart =
+    private static void assertThatHasMultiPartUploadWithPart(
```

Comment:
Both custom assert method with a for-loop can be simplified to something like this:
```java
private static void assertThatHasUploadedObject(StubMultiPartUploader actual, byte[] content) {
    TestPutObjectResult expectedIncompletePart =
            createPutObjectResult(TEST_OBJECT_NAME, content);

    assertThat(actual.getIncompletePartsUploaded()).contains(expectedIncompletePart);
}
```